.github/actions/changes/action.yml

```diff
@@ -0,0 +1,52 @@
+name: "Github Changes"
+description: "Compute by any means necessary the changes attempting to be merged, or that have been merged in to a branch."
+inputs:
+  workflow-file:
+    description: "local repository path to the workflow file, no leading slash."
+    required: true
+outputs:
+  changes-base-githash:
+    description: The shared git hash of a pr, a bors target branch, or the last successful build of the workflow-file on the branch.
+    value: ${{ steps.calc.outputs.changes-base-githash }}
+  changes-target-branch:
+    description: If a target branch exists for either bors or a pr, the branch.
+    value: ${{ steps.calc.outputs.changes-target-branch }}
+  changes-pull-request-number:
+    description: If a pullrequest number can be found, either in bors, or in a pr, the number.
+    value: ${{ steps.calc.outputs.changes-pull-request-number }}
+runs:
+  using: "composite"
+  steps:
+    - id: calc
+      run: |
+        if [[ $(git log --pretty=format:"%H" -n 26 | wc -l) < 2 ]]; then
+          echo It is likely that checkout did not return more than one commit, recommend calling checkout with fetch-depth: 25 1>&2
+          echo BASE_GITHASH is likely to not be found even if it would otherwise be the immediate prior revision. 1>&2
+        fi
+
+        eval $(./.github/actions/changes/get_pr_info.sh -w "${WORKFLOW}" -b -d)
+
+        echo "CHANGES_TARGET_BRANCH=${CHANGES_TARGET_BRANCH}" >> $GITHUB_ENV
```

Comment:
You can expose both environment variables to future builds in a job, and expose a variable as action output.   I think it's smart to do this in general to allow for simpler files as long as we document the behavior.